app/main.py

```diff
@@ -1,3 +1,42 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    __slots__ = "km"
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        return Distance(self.km + getattr(other, "km", other))
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        self.km += getattr(other, "km", other)
+        return self
+
+    def __mul__(self, other: float | int) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: float | int) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: float | int) -> bool:
+        return self.km < getattr(other, "km", other)
+
+    def __gt__(self, other: float | int) -> bool:
+        return self.km > getattr(other, "km", other)
```

Comment:
This is clever  👏 . But it also means that your code will accept any object that has "km" attribute while we want to compare only  between numeric data types and 'Distance' objects. Also, type annotations are missing.

Reply:
I see that you added the types 👍. 
However right now your code will perform comparison with any data type that will have a 'km' attribute and this may lead to unexpected beahviour. E.g
```
class NotADistance:
    def __init__(self, km: int) -> None:
        self.km = 1
``` 
will be considered a valid class for comparison operation while it should not be.